### ISSUE TITLE
Search repo from server

### ISSUE THREAD

**Issue by unional** [thedillonb/CodeHub#166]
Currently the search function in repositories (at least, for what I know) only search from fetched results. It doesn't work correctly if the number of repositories > fetch limit.

**Comment by thedillonb** [comment]
@unional, you're right. Searching doesn't generally go to the server to get results. Instead it tries to filter locally. There doesn't actually exist an API that provides "search" functionality for listing repositories (except when searching globally across GitHub). However, you bring up an interesting bug in that searching should search locally, reduce the list of items, and then, if there are few matches, paginate the remaining items from the server and search through those - repeating the process. Unfortunately, it looks like the app is not doing that. However, we can fit it!
